pkg/config/util.go

```diff
@@ -0,0 +1,130 @@
+package config
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"io"
+	"math/big"
+	"strings"
+
+	acsapi "github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/tls"
+	"golang.org/x/crypto/bcrypt"
+	"k8s.io/client-go/tools/clientcmd/api/v1"
+)
+
+func makeKubeConfig(clientKey *rsa.PrivateKey, clientCert, caCert *x509.Certificate, endpoint, username, namespace string) (*v1.Config, error) {
+	clustername := strings.Replace(endpoint, ".", "-", -1)
+	authinfoname := username + "/" + clustername
+	contextname := namespace + "/" + clustername + "/" + username
+
+	caCertBytes, err := tls.CertAsBytes(caCert)
+	if err != nil {
+		return nil, err
+	}
+	clientCertBytes, err := tls.CertAsBytes(clientCert)
+	if err != nil {
+		return nil, err
+	}
+	clientKeyBytes, err := tls.PrivateKeyAsBytes(clientKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1.Config{
+		APIVersion: "v1",
+		Kind:       "Config",
+		Clusters: []v1.NamedCluster{
+			{
+				Name: clustername,
+				Cluster: v1.Cluster{
+					Server: "https://" + endpoint,
+					CertificateAuthorityData: caCertBytes,
+				},
+			},
+		},
+		AuthInfos: []v1.NamedAuthInfo{
+			{
+				Name: authinfoname,
+				AuthInfo: v1.AuthInfo{
+					ClientCertificateData: clientCertBytes,
+					ClientKeyData:         clientKeyBytes,
+				},
+			},
+		},
+		Contexts: []v1.NamedContext{
+			{
+				Name: contextname,
+				Context: v1.Context{
+					Cluster:   clustername,
+					Namespace: namespace,
+					AuthInfo:  authinfoname,
+				},
+			},
+		},
+		CurrentContext: contextname,
+	}, nil
+}
+
+func makeHtPasswd(username, password string) ([]byte, error) {
+	b, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return nil, err
+	}
+
+	return append([]byte(username+":"), b...), nil
+}
+
+func randomBytes(n int) ([]byte, error) {
+	b := make([]byte, n)
+	if _, err := io.ReadAtLeast(rand.Reader, b, n); err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+func randomStorageAccountName() (string, error) {
+	const letterBytes = "abcdefghijklmnopqrstuvwxyz0123456789"
+
+	b := make([]byte, 24)
+	for i := range b {
+		o, err := rand.Int(rand.Reader, big.NewInt(int64(len(letterBytes))))
+		if err != nil {
+			return "", err
+		}
+		b[i] = letterBytes[o.Int64()]
+	}
+
+	return string(b), nil
+}
+
+func randomString(length int) (string, error) {
+	const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+	b := make([]byte, length)
+	for i := range b {
+		o, err := rand.Int(rand.Reader, big.NewInt(int64(len(letterBytes))))
+		if err != nil {
+			return "", err
+		}
+		b[i] = letterBytes[o.Int64()]
+	}
+
+	return string(b), nil
+}
+
+func selectDNSNames(cs *acsapi.OpenShiftManagedCluster) {
```

Comment:
this function doesn't belong here

Reply:
it should probably be refactored away

Reply:
@jim-minter , another place for this to be split would be a template and be using goTemplate. I would like to avoid constructing variables in that place and keep all config generation in here. What you had in mind for this?

Reply:
I think RouterLBCNamePrefix, MasterLBCNamePrefix and other derived configs should probably be member functions on OpenShiftManagedCluster.
I don't know what the rationale is for the rest of the function - why is it there?